src/commands/image/imageSource/buildImageInAzure/RunStep.ts

```diff
@@ -28,10 +30,18 @@ export class RunStep extends ExecuteActivityOutputStepBase<BuildImageInAzureImag
                 dockerFilePath: path.basename(context.dockerfilePath) /* Assume the dockerfile is always in the root of the source */
             };
 
-            const building: string = localize('buildingImage', 'Building image...');
-            progress.report({ message: building });
-
-            context.run = await context.client.registries.beginScheduleRunAndWait(context.resourceGroupName, context.registryName, runRequest);
+            const retries = 3;
+            await retry(
+                async (currentAttempt: number) => {
+                    context.run = await context.client.registries.beginScheduleRunAndWait(context.resourceGroupName, context.registryName, runRequest);
+                    const message: string = currentAttempt === 1 ?
+                        localize('buildingImage', 'Building image...') :
+                        localize('buildingImageAttempt', 'Building image (Attempt {0}/{1})...', currentAttempt, retries + 1);
```

Comment:
Why's it retries + 1?

Reply:
Techinically the first try is not a retry, so if there are 3 retries allowed that is technically the fourth try. If you think we should only do 3 tries I can change the retries to 2.

Reply:
Oh yeah, makes sense. I feel like 3 is a good number to me.